pkg/collector/upgrade/v_0_122_0.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package upgrade
+
+import (
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1beta1"
+)
+
+func upgrade0_122_0(u VersionUpgrade, otelcol *v1beta1.OpenTelemetryCollector) (*v1beta1.OpenTelemetryCollector, error) { //nolint:unparam
+	tel := otelcol.Spec.Config.Service.GetTelemetry()
+
+	// if tel == nil || tel.Metrics.Address == "" {
+	// 	err := otelcol.Spec.Config.Service.ApplyDefaults(u.Log)
+	// 	return otelcol, err
+	// }
```

Comment:
Here, I think we should just return.

```suggestion
	if tel == nil || tel.Metrics.Address == "" {
	 	return otelcol, err
	 }
```

Reply:
sorry. commented to test and forget to remove